beacon-chain/cache/sync_committee_head_state.go

```diff
@@ -25,23 +25,33 @@ func NewSyncCommitteeHeadState() *SyncCommitteeHeadStateCache {
 }
 
 // Put `slot` as key and `state` as value onto the cache.
-func (c *SyncCommitteeHeadStateCache) Put(slot types.Slot, st state.BeaconState) {
+func (c *SyncCommitteeHeadStateCache) Put(slot types.Slot, st state.BeaconState) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
+	// Make sure that the provided state is non nil
+	// and is of the correct type.
+	if st == nil || st.IsNil() {
+		return ErrNilState
```

Comment:
Let's use more general package-level errors here. For example, ErrNilValueProvided, which can be reused across caches

Reply:
I like the idea of typed errors in the cache package to help us get to caches all sharing the same semantics :+1: